data_structures/arrays/get_products_of_all_other_elements.rb

```diff
@@ -85,3 +111,53 @@ def products(nums)
 
 puts(products([1, 2, 3]))
 # => [6, 3, 2]
+
+#
+# Approach 3: O(1) space approach
+#
+
+# This approach is essentially an extension of the approach 2.
+# Basically, we will be using the output array as one of L or R and we will
+# be constructing the other one on the fly.
+
+# Complexity analysis
+#
+# Time complexity: O(N) where N represents the number of elements in the input
+# array. We use one iteration to construct the array L, one to update the array
+# answer.
+
+# Space complexity: O(1) since don't use any additional array for our
+# computations. The problem statement mentions that using the answer
+# array doesn't add to the space complexity.
+
+def products(nums)
+  return [] if nums.size < 2
```

Comment:
(This may be out of scope of this PR.)

This file uses all of `Array#count`, `Array#length` and `Array#size` to get the number of elements and looks inconsistent.

Reply:
kk